onnxruntime/core/providers/webnn/builders/impl/rotaryEmbedding_op_builder.cc

```diff
@@ -276,12 +285,19 @@ Status RotaryEmbeddingOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_build
     output = wnn_builder.call<emscripten::val>("concat", concat_inputs, 3, concat_back_input_options);
   }
 
-  // Reshape the output to the original shape. The output shape is the same as the input shape.
-  const std::vector<uint32_t> output_shape = GetNarrowedIntfromInt64<uint32_t>(input_shape);
-  emscripten::val reshape_output_options = emscripten::val::object();
-  reshape_output_options.set("label", node_name + "_reshape_output");
-  output = wnn_builder.call<emscripten::val>(
-      "reshape", output, emscripten::val::array(output_shape), reshape_output_options);
+  if (input_is_4d) {
+    // The output is in 4D shape, we need to transpose it back to the original shape.
+    transpose_options.set("label", node_name + "_transpose_output");
```

Comment:
Propose adding a clarifying a comment that we don't need to compute an inverse permutation here because the original permutation also *happens* to be its own inverse. `inverse({0, 2, 1, 3}) == {0, 2, 1, 3}`

Reply:
Good point, done.